### PR TITLE
Slight code improvements.

### PR DESCRIPTION
**Pull request by apintea**
1) Treated errors;
2) Changed the import dependencies to be local;
3) Added comments.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -6,52 +6,65 @@ import (
 	"path/filepath"
 	"flag"
 
-	"github.com/docker/docker/builder/dockerignore"
-	"github.com/docker/docker/pkg/fileutils"
+	"./dockerignore"
```
Comment (rnc) [inline main.go]: Why commit the docker code locally?

File: main.go [hunk 2]
```diff
@@ -6,52 +6,65 @@ import (
...
+	// Walk the file tree rooted at root, calling cleanUp() for each file or directory in the tree, including root.
+	filepath.Walk(root, cleanUp)
 }
 
-func check(e error) {
-    if e != nil {
-        panic(e)
+func check(err error) {
+    // If error is not Null
+    if err != nil {
```
Comment (rnc) [inline main.go]: What is the point of this comment ? Comments should describe the "why the code is doing" ; err is != null is self-explanatory

File: main.go (hunk 1 above)
Comment (apintea) [reply main.go]: Isn't it a good practice to have all the dependencies locally for the RPM, rather than pointing to Github?

File: main.go [hunk 3]
```diff
@@ -6,52 +6,65 @@ import (
...
-	root := flag.Arg(0)
-	if len(root) == 0 {
-		fmt.Printf ("Pass in file path to examine.\n")
-		panic("Missing main argument")
-	}
+	// Check the number of command-line arguments.
+	// It should be 2: the .go file itslef and the path to inspect.
+	if len(os.Args) != 2 {
+		panic("Please provide a valid path to be inspected.")
+	}
```
Comment (rnc) [inline main.go]: Was the len(root) test not sufficient?

File: main.go [hunk 4]
```diff
@@ -6,52 +6,65 @@ import (
...
-	check(dockerignoreerr)
+	// Dynamically assign the passed argument for further inspection.
+	root := flag.Arg(0)
+	
+	// Print out the path on which .dockerignore will be applied.
+	fmt.Println("Will apply .dockerignore on path", root)
+	// Opens the .dockerignore file for reading.
+	file, err := os.Open(".dockerignore")
+	// Checks if there is an error.
+	check(err)
```
Comment (rnc) [inline main.go]: Can we try to minimise the change-set ; it seems like there is variable renaming for no apparent reason

File: main.go [hunk 5]
```diff
@@ -6,52 +6,65 @@ import (
...
 
-	excludes, _ = dockerignore.ReadAll(file)
-	fmt.Printf("Patterns are %v\n",excludes)
+	// Read and assign the content which needs to be excluded.
+	excludes, err = dockerignore.ReadAll(file)
+	// Check if there is an error.	
+	check(err)
```
Comment (rnc) [inline main.go]: Lots of these comments are superfluous.

File: main.go [hunk 6]
```diff
@@ -6,52 +6,65 @@ import (
...
+func check(err error) {
+    // If error is not Null
+    if err != nil {
+    // Print the error message and stop
+        panic(err)
     }
 }
 
-func isDirectory(path string) string {
-	fileInfo, err := os.Stat(path)
```
Comment (rnc) [inline main.go]: Why remove this?

File: main.go [hunk 7]
```diff
@@ -6,52 +6,65 @@ import (
...
+// Check if a files name is equivalent with one of the Strings from the excludes array. If so, remove it.
+func cleanUp(file string, f os.FileInfo, err error) error {
+	// Check if the name of the file matches with a String from the excludes array.
```
Comment (rnc) [inline main.go]: Why rename this? These are unnecessary changes

File: main_test.go [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+func TestCleanUp(t *testing.T) {
+
```
Comment (rnc) [inline main_test.go]: Er, where is the test? ;-)

File: main.go [hunk 9]
```diff
@@ -6,52 +6,65 @@ import (
...
-	}
-}
-
-func visit(path string, f os.FileInfo, err error) error {
-	rm, _ := fileutils.Matches(path, excludes)
+	// rm is of type boolean. If the files name was found in the excludes array
 	if rm {
-		fmt.Printf("Removing %v %v\n", isDirectory(path), path)
-		var err = os.Remove(path)
+		// Print out the name of the file that's removed
```
Comment (rnc) [inline main.go]: This comment is wrong anyway. It might not be a file but a directory

**Comment by rnc** [comment]
I've added a lot of comments. I am not sure about a lot of these changes. Can we minimise this to just what is required i.e. the tests and the missing error code checks?

File: main.go (hunk 1 above)
Comment (rnc) [reply main.go]: What has the RPM got to do with the Go library? I *thought* that layout was standard for Go

File: main.go [hunk 10]
```diff
@@ -6,52 +6,65 @@ import (
...
 
-	cwd, _ := os.Getwd()
-	fmt.Printf("Will apply .dockerignore on path %v ( %v )\n", root, cwd)
-	file, dockerignoreerr := os.Open(".dockerignore")
-	check(dockerignoreerr)
+	// Dynamically assign the passed argument for further inspection.
+	root := flag.Arg(0)
+	
+	// Print out the path on which .dockerignore will be applied.
+	fmt.Println("Will apply .dockerignore on path", root)
```
Comment (rnc) [inline main.go]: Why change the logging statement? 
(Minor) Why use println instead of printf ?

File: main.go (hunk 5 above)
Comment (rnc) [reply main.go]: With first read, because of the extraneous changes I missed the vital requested change to actually check the error i.e. removing `_` and replacing with `err`. If the PR could be trimmed down to just these sort of changes it would be much easier to review.

File: main.go (hunk 1 above)
Comment (apintea) [reply main.go]: dockerignore and fileutils are 2 packages that are on Github. They are actually 2 .go files. I simply put them in the local dir tree, in order to not point to Github at all. Isn't it a mistake to point to a remote library (and not Red Hat) when building?

File: main.go (hunk 2 above)
Comment (apintea) [reply main.go]: Comments are very useful even in this case, when learning a not so readable language as Go! But I will remove them. !=null is self-explanatory, but !=**nil** not so... yuck! :-D

File: main.go (hunk 3 above)
Comment (apintea) [reply main.go]: The root variable was assigned before checking if the source will even provide a value! :-)
I will modify as it was.

File: main.go (hunk 4 above)
Comment (apintea) [reply main.go]: What do you mean "variable renaming for no apparent reason"?

File: main.go (hunk 5 above)
Comment (apintea) [reply main.go]: OK!

File: main.go (hunk 6 above)
Comment (apintea) [reply main.go]: Seems like I've had another version of this repo :-(

File: main.go (hunk 7 above)
Comment (apintea) [reply main.go]: Will revert.

File: main.go (hunk 1 above)
Comment (rnc) [reply main.go]: We don't need them and shouldn't copy them in the upstream repository.

File: main.go (hunk 4 above)
Comment (rnc) [reply main.go]: I meant as part of line 25 where the variable was renamed.

File: main.go (hunk 9 above)
Comment (apintea) [reply main.go]: Hey, https://en.wikipedia.org/wiki/Directory_(computing)
"**_file_** system" :-D

File: main.go (hunk 10 above)
Comment (apintea) [reply main.go]: Will inspect this.
